pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -48,3 +77,41 @@ func (pc *PackageControllerClient) ValidateControllerDoesNotExist(ctx context.Co
 	}
 	return nil
 }
+
+func (pc *PackageControllerClient) ApplySecret(ctx context.Context) error {
+	eksaAccessKeyId, eksaSecretAccessKey := os.Getenv(config.EksaAccessKeyIdEnv), os.Getenv(config.EksaSecretAcessKeyEnv)
+	eksaRegion := eksaDefaultRegion
+	if region, found := os.LookupEnv(config.EksaRegionEnv); found {
+		eksaRegion = region
+	}
+
+	templateValues := map[string]string{
+		"eksaAccessKeyId":     base64.StdEncoding.EncodeToString([]byte(eksaAccessKeyId)),
+		"eksaSecretAccessKey": base64.StdEncoding.EncodeToString([]byte(eksaSecretAccessKey)),
+		"eksaRegion":          base64.StdEncoding.EncodeToString([]byte(eksaRegion)),
+	}
+
+	result, err := templater.Execute(awsSecretYaml, templateValues)
+	if err != nil {
+		return fmt.Errorf("replacing template values %v", err)
+	}
```

Comment:
nit: I would just define the secret using go structs and just marshal it to yaml. I think it makes the code a bit more clear since you don't need to go read the template yaml file and you have everything defined here in code